src/main/java/org/spongepowered/api/schematic/Schematic.java

```diff
@@ -0,0 +1,202 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.schematic;
+
+import com.flowpowered.math.vector.Vector3i;
+import com.google.common.collect.BiMap;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.api.world.World;
+
+import java.util.Optional;
+
+/**
+ * Unless otherwise noted, "the specification" refers to the latest version of the Schematic Specification {@url https://github
+ * .com/SpongePowered/Schematic-Specification}.
+ */
+public interface Schematic extends DataSerializable {
+
+    /**
+     * Gets the format version used. Correlates to the `Version` field of the specification.
+     *
+     * @return The format version used
+     */
+    int getVersion();
+
+    /**
+     * Gets the attached {@link DataContainer}. Correlates to the optional `Metadata` field of the specification.
+     *
+     * @return The attached {@link DataContainer} or {@link Optional#empty()} if none
+     */
+    Optional<DataContainer> getMetadata();
+
+    /**
+     * Gets the width of the Schematic. Correlates to the `Width` field of the specification.
+     *
+     * @return The width of the Schematic.
+     */
+    int getWidth();
+
+    /**
+     * Gets the height of the Schematic. Correlates to the `Height` field of the specification.
+     *
+     * @return The height of the Schematic.
+     */
+    int getHeight();
+
+    /**
+     * Gets the length of the Schematic. Correlates to the `Length` field of the specification.
+     *
+     * @return The length of the Schematic.
+     */
+    int getLength();
+
+    /**
+     * Gets the origin of the Schematic (relative to the minimum point) or the minimum point if none. Correlates to the `OffsetX`, `OffsetY`, and
+     * `OffsetZ` fields of the specification.
+     *
+     * @return The origin of the Schematic
+     */
+    Vector3i getOrigin();
```

Comment:
The Offset fields define an origin for recreating the build. Might as well be a Vector3i, because it'd just be three int fields.